Modules/grpmodule.c

```diff
@@ -281,23 +280,33 @@ static PyObject *
 grp_getgrall_impl(PyObject *module)
 /*[clinic end generated code: output=585dad35e2e763d7 input=d7df76c825c367df]*/
 {
-    PyObject *d;
-    struct group *p;
-
-    if ((d = PyList_New(0)) == NULL)
+    PyObject *d = PyList_New(0);
+    if (d == NULL) {
         return NULL;
+    }
+
+    static PyMutex getgrall_mutex = {0};
```

Comment:
I'm not sure if `PyMutex` likes being const, but this should fix the C analyzer.
```suggestion
    static const PyMutex getgrall_mutex = {0};
```

Reply:
No, it should not be const.

Reply:
Ok, we'll need to do something about the analyzer then if your plan is to use this pattern for other thread-unsafe libc functions. I can probably submit a PR for that sometime today. @vstinner, in the meantime, add this to `ignored.tsv` to fix CI.

Reply:
> @vstinner, in the meantime, add this to ignored.tsv to fix CI.